fedora/clients/fasjson.py

```diff
@@ -101,3 +101,51 @@ async def get_users_by_matrix_id(self, matrix_id: str) -> dict | str:
             )
 
         return searchresult[0]
+
+    async def get_user_groups(self, username, params=None):
+        """
+        Retrieves all groups a user belongs to, including membership type (member or sponsor).
+
+        Args:
+            username (str): Username of the user to query.
+            params (dict, optional): Additional query parameters to include in the request.
+
+        Returns:
+            List[dict] or None:
+                - A list of dictionaries containing group information:
+                    - groupname (str): Name of the group.
+                    - membership_type (str): "member" or "sponsor" depending on the user's role.
+                - None (if the user is not found or there's an error).
+
+        Raises:
+            InfoGatherError: If there's an error fetching data from Fasjson or if the user is not found (404).
+        """
+
+        try:
+            response = await self._get("/".join(["users", username, "groups"]), params=params)
+            user_groups = response.json().get("groups", [])
+
+            group_details = []
+            for groupname in user_groups:
+                # Fix - i realized at some point that get group membership, just returns members/sponsors in a group
```

Comment:
This looks like maybe a note for yourself during development, I think you can drop this.

Reply:
> This looks like maybe a note for yourself during development, I think you can drop this.

Yes it was haha, forgot to take that out. Thank you